### ISSUE TITLE
[Bug] Do data backup at all

### ISSUE THREAD

**Issue by rysson** [machiav3lli/oandbackupx#376]
**Description**
There is no data backup for any app. Only apk are backed up.

**Steps To Reproduce**

1. Select any user app (main tab)
2. Create backup
3. Select all (app + data)
4. Only app is backed up.

**Expected behavior**
App data should be backed up.

**Screenshots**
[a01c.log](https://github.com/machiav3lli/oandbackupx/files/6832233/a01c.log)

Only if there's no clear logs, add screenshots to help explain your problem.

**System Information(please complete the following information):**

- Device: Huawei mate 10 pro
- Android Version: 8.0.0
- ROM: stock
- App's Version: 6.0.1

**Additional Notes**

There is no matter if I try to back up single app or all apps. There is no any data backup (for 300+ apps).
If I choose only data (no app) there is empty backup created.
Teere is no matter to select data, private date, external data, obb data or any combination.

**Comment by Gaston48** [comment]
I have the same issue: apk-backup works but no data backup possible
Any data-backup-intent ends with nullPointerException: null

`
adb shell su -c /system/bin/toybox ls -AllZ /data/user/0/inc.flide.vi8` 
_`ls:_ /data/user/0/inc.flide.vi8: No such file or _directory`_

this happens with any app. 

-  LineageOs 17.1.
- Root (Magisk)
- OandBackup 7.0.0

Thanks for help!

**Comment by hg42** [comment]
I now learned that su needs a --mount-master option on newer Android versions (API 30++).
Otherwise the namespace of the OAndBackupX user id is totally different from that of any other app and even root doesn't see the files of other apps.

```
adb shell su --mount-master -c /system/bin/toybox ls -AllZd /data/user/0/inc.flide.vi8
adb shell su --mount-master -c /system/bin/toybox ls -AllZ /data/user/0/inc.flide.vi8
```
Depending on the way you become superuser in a terminal, you might also need to use this.

**Comment by hg42** [comment]
OABX should use this flag in the configuration, but I'm not sure if that was always the case

**Comment by machiav3lli** [comment]
@hg42 it's already is. In ShellHandler.kt you can see the flag `Shell.FLAG_MOUNT_MASTER` added at shell building.

**Comment by hg42** [comment]
well, that wasn't clear language from my side:

> OABX should use this flag in the configuration

= I saw it in the  configuration of libsu and it should work, though I didn't test

> but I'm not sure if that was always the case

= it's there, but I didn't elaborate, since when

**Comment by hg42** [comment]
I think this should be fixed with #493
